sled-agent/src/instance_manager.rs

```diff
@@ -87,97 +87,121 @@ impl InstanceManager {
+                    info!(&self.inner.log,
+                          "instance already registered with another Propolis ID";
+                          "instance_id" => %instance_id,
+                          "existing_propolis_id" => %*existing_propolis_id);
+                    return Err(Error::Instance(
+                        crate::instance::Error::InstanceAlreadyRegistered(
+                            *existing_propolis_id,
+                        ),
+                    ));
+                } else {
                     info!(
                         &self.inner.log,
-                        "instance already exists with given dst propolis"
+                        "instance already registered with requested Propolis ID"
                     );
-                    (instance.clone(), None)
+                    existing_instance.clone()
                 }
-                _ => {
-                    // Instance does not exist or one does but we're performing
-                    // a intra-sled migration. Either way - create an instance
-                    info!(&self.inner.log, "new instance");
-                    let instance_log = self.inner.log.new(o!());
-                    let instance = Instance::new(
-                        instance_log,
-                        instance_id,
-                        initial_hardware,
-                        self.inner.vnic_allocator.clone(),
-                        self.inner.port_manager.clone(),
-                        self.inner.lazy_nexus_client.clone(),
-                    )?;
-                    let instance_clone = instance.clone();
-                    let old_instance = instances
-                        .insert(instance_id, (target_propolis_id, instance));
-                    if let Some((_old_propolis_id, old_instance)) = old_instance
-                    {
-                        // If we had a previous instance, we must be migrating
-                        assert!(migrate.is_some());
-                        // TODO: assert that old_instance.inner.propolis_id() == migrate.src_uuid
-
-                        // We forget the old instance because otherwise if it is the last
-                        // handle, the `InstanceTicket` it holds will also be dropped.
-                        // `InstanceTicket::drop` will try to remove the corresponding
-                        // instance from the instance manager. It does this based off the
-                        // instance's ID. Given that we just replaced said instance using
-                        // the same ID, that would inadvertantly remove our newly created
-                        // instance instead.
-                        // TODO: cleanup source instance properly
-                        std::mem::forget(old_instance);
-                    }
+            } else {
+                info!(&self.inner.log, "new instance");
+                let instance_log = self.inner.log.new(o!());
```

Comment:
Mind adding `instance_id` to the log context here

Reply:
Done in 2067fc1a3.